### PR TITLE
Add Databricks Periodic Integration tests

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Add Databricks Periodic Integration tests
## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: .github/workflows/spark-periodic-tests.yml
Comment (kenxu95) [inline .github/workflows/spark-periodic-tests.yml:1]: Curious why we separated this from the other periodic tests?

**Review by kenxu95 — APPROVED** [review]
Looks great! Just one question wondering why we don't merge this into `periodic-integration-tests.yml`

File: .github/workflows/spark-periodic-tests.yml
Comment (kenxu95) [inline .github/workflows/spark-periodic-tests.yml:33]: Also, do things fail if we crank the concurrency up? It might make things a lot faster (you may have to provision a slightly beefier machine though).

File: .github/workflows/spark-periodic-tests.yml
Comment (hsubbaraj-spiral) [reply .github/workflows/spark-periodic-tests.yml:1]: I did this originally because of the different schedule, the spark tests run once a week vs every day (due to machine cost of running the databricks tests.)

File: .github/workflows/spark-periodic-tests.yml
Comment (hsubbaraj-spiral) [reply .github/workflows/spark-periodic-tests.yml:33]: Will experiment with this before merging

File: .github/workflows/spark-periodic-tests.yml
Comment (kenxu95) [reply .github/workflows/spark-periodic-tests.yml:1]: ah I see. What's the cost for each run? I wonder if we should do twice a week at least (eg. tuesday and friday)

**Pull request merged**
